packages/api-clients/open-api/m2mGatewayApi.yml

```diff
@@ -3225,6 +3225,87 @@ paths:
               $ref: "#/components/headers/RateLimitRemainingHeader"
             X-Rate-Limit-Interval:
               $ref: "#/components/headers/RateLimitIntervalHeader"
+    delete:
+      tags:
+        - purposes
+      summary: Delete Purpose
+      description: Delete Purpose. Only purposes with one version in Draft or WaitingForApproval state can be deleted
+      operationId: deletePurpose
+      responses:
+        "204":
+          description: Purpose deleted
```

Comment:
Waiting a clarification on this topic, 200 vs 204 in delete operations: https://pagopaspa.slack.com/archives/C08J74Y0H96/p1751618163158319